File: ios/bridging_oldArch/THEORCTContentProtectionModule.m
//
//  THEORCTContentProtectionModule.m
//

#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(THEORCTContentProtectionModule, NSObject)

RCT_EXTERN_METHOD(onBuildProcessed:(NSDictionary)result);

RCT_EXTERN_METHOD(onCertificateRequestProcessedAsRequest:(NSDictionary)result);

RCT_EXTERN_METHOD(onCertificateRequestProcessedAsCertificate:(NSDictionary)result);

RCT_EXTERN_METHOD(onCertificateResponseProcessed:(NSDictionary)result);

RCT_EXTERN_METHOD(onLicenseRequestProcessedAsRequest:(NSDictionary)result);

RCT_EXTERN_METHOD(onLicenseRequestProcessedAsLicense:(NSDictionary)result);

RCT_EXTERN_METHOD(onLicenseResponseProcessed:(NSDictionary)result);

RCT_EXTERN_METHOD(onExtractFairplayContentIdProcessed:(NSDictionary)result);

RCT_EXTERN_METHOD(registerContentProtectionIntegration:(nonnull NSString *)integrationId
                  keySystemId:(nonnull NSString *)keySystemId);

@end

File: ios/bridging_oldArch/THEORCTContentProtectionModule.swift
//
//  THEORCTContentProtectionModule.swift
//

@objc(THEORCTContentProtectionModule)
class THEORCTContentProtectionModule: RCTEventEmitter {
    let contentProtectionAPI = THEOplayerRCTContentProtectionAPI()
    
    override init() {
        super.init()
        self.contentProtectionAPI.sendEvent = self.sendEvent
    }
    
    override static func moduleName() -> String! {
        return "THEORCTContentProtectionModule"
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    override func supportedEvents() -> [String]! {
        return ["onBuildIntegration",
                "onCertificateRequest",
                "onCertificateResponse",
                "onLicenseRequest",
                "onLicenseResponse",
                "onExtractFairplayContentId"
        ]
    }
    
    @objc(registerContentProtectionIntegration:keySystemId:)
    func registerContentProtectionIntegration(_ integrationId: String, keySystemId: String) -> Void {
        self.contentProtectionAPI
            .registerContentProtectionIntegration(
                integrationId: integrationId,
                keySystemId: keySystemId
            )
    }
    
    @objc(onBuildProcessed:)
    func onBuildProcessed(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onBuildProcessed(result: result)
    }
    
    @objc(onCertificateRequestProcessedAsRequest:)
    func onCertificateRequestProcessedAsRequest(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onCertificateRequestProcessedAsRequest(result: result)
    }
    
    @objc(onCertificateRequestProcessedAsCertificate:)
    func onCertificateRequestProcessedAsCertificate(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onCertificateRequestProcessedAsCertificate(result: result)
    }
    
    @objc(onCertificateResponseProcessed:)
    func onCertificateResponseProcessed(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onCertificateResponseProcessed(result: result)
    }
    
    @objc(onLicenseRequestProcessedAsRequest:)
    func onLicenseRequestProcessedAsRequest(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onLicenseRequestProcessedAsRequest(result: result)
    }
    
    @objc(onLicenseRequestProcessedAsLicense:)
    func onLicenseRequestProcessedAsLicense(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onLicenseRequestProcessedAsLicense(result: result)
    }
    
    @objc(onLicenseResponseProcessed:)
    func onLicenseResponseProcessed(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onLicenseResponseProcessed(result: result)
    }
    
    @objc(onExtractFairplayContentIdProcessed:)
    func onExtractFairplayContentIdProcessed(_ result: NSDictionary) -> Void {
        self.contentProtectionAPI.onExtractFairplayContentIdProcessed(result: result)
    }
}
